packages/std/src/imports.rs

```diff
@@ -78,20 +78,20 @@ impl ReadonlyStorage for ExternalStorage {
             (None, None) => unsafe { db_scan(0, 0, order) },
             (Some(s), None) => {
                 let start = build_region(s);
-                let start_ptr = &*start as *const Region as u32;
-                unsafe { db_scan(start_ptr, 0, order) }
+                let start_ptr = start.as_ref() as *const Region;
```

Comment:
I think this way we can easily run into the same issue we had in the beginning: the box and reference to it can be destroyed after the pointer way copied. On order to avoid this, I'd store the reference in a variable and cast to ` as *const Region  as u32` on demand.

Reply:
I did it this way for "clarity", but what you're saying makes a lot of sense: in this way, we're basically storing a raw pointer, which can become dissasociated with the original value. And Rust wouldn't complain, because it's a raw thingy.